server/src/main/java/io/druid/segment/realtime/firehose/ServiceAnnouncingChatHandlerProvider.java

```diff
@@ -52,24 +54,36 @@ public ServiceAnnouncingChatHandlerProvider(
     this.node = node;
     this.serviceAnnouncer = serviceAnnouncer;
     this.handlers = Maps.newConcurrentMap();
+    this.announcements = new ConcurrentSkipListSet<>();
   }
 
   @Override
   public void register(final String service, ChatHandler handler)
   {
-    final DruidNode node = makeDruidNode(service);
+    register(service, handler, true);
+  }
+
+  @Override
+  public void register(final String service, ChatHandler handler, boolean announce)
+  {
     log.info("Registering Eventhandler[%s]", service);
 
     if (handlers.putIfAbsent(service, handler) != null) {
       throw new ISE("handler already registered for service[%s]", service);
     }
 
-    try {
-      serviceAnnouncer.announce(node);
-    }
-    catch (Exception e) {
-      log.warn(e, "Failed to register service[%s]", service);
-      handlers.remove(service, handler);
+    if (announce)
+    {
+      try {
+        serviceAnnouncer.announce(makeDruidNode(service));
+        if (!announcements.add(service)) {
```

Comment:
ISE probably makes more sense; if you had identical service names you could get this to happen, and if we just logged and carried on there's a chance you'd get things talking to the wrong things thinking that they were talking to other things.